extensions/resteasy-classic/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java

```diff
@@ -176,10 +176,10 @@ JaxrsProvidersToRegisterBuildItem setupProviders(BuildProducer<ReflectiveClassBu
             }
         }
         contributedProviders.addAll(annotatedProviders);
+        // this one is added manually in RESTEasy's ResteasyDeploymentImpl
+        contributedProviders.add(ServerFormUrlEncodedProvider.class.getName());
```

Comment:
See my other comment. You're changing the semantic of `contributedProviders` by adding this class in it (it's one that is registered by RESTEasy, not a contributed one) and this is problematic IMO. It makes things extremely confusing.